src/main/java/me/totalfreedom/totalfreedommod/command/Command_spectator.java

```diff
@@ -6,16 +6,39 @@
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
-@CommandPermissions(level = Rank.SUPER_ADMIN, source = SourceType.ONLY_IN_GAME)
-@CommandParameters(description = "Quickly change your own gamemode to spectator.", usage = "/<command>", aliases = "gmsp")
+@CommandPermissions(level = Rank.SUPER_ADMIN, source = SourceType.BOTH)
+@CommandParameters(description = "Quickly change your own gamemode to spectator, or define someone's username to change theirs.", usage = "/<command> <-a | [partialname]>", aliases = "gmsp")
 public class Command_spectator extends FreedomCommand
 {
-
-    @Override
-    public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
-    {
-        playerSender.setGameMode(GameMode.SPECTATOR);
-        msg("Gamemode set to spectator.");
-        return true;
+    public boolean run(final CommandSender sender, final Player playerSender, final Command cmd, final String commandLabel, final String[] args, final boolean senderIsConsole) {
+        if (args.length == 0) {
+            if (this.isConsole()) {
+                sender.sendMessage("When used from the console, you must define a target player.");
+                return true;
+            }
+            this.checkRank(Rank.SUPER_ADMIN);
+            playerSender.setGameMode(GameMode.SPECTATOR);
+            this.msg("Gamemode set to Spectator.");
+            return true;
+        }
+        else {
+            this.checkRank(Rank.SUPER_ADMIN);
```

Comment:
Again, I dont understand why you are checking if they are a super admin when you have to be a super admin to even use the command? Not to mention with this I would not even like the idea of setting everyones rank to Spectator, much less by a super admin...